src/control/configureIndex.ts

```diff
@@ -4,54 +4,50 @@ import {
   ConfigureIndexRequest,
 } from '../pinecone-generated-ts-fetch/control';
 import { PineconeArgumentError } from '../errors';
-import { buildValidator } from '../validator';
 import type { IndexName } from './types';
-
-import { Type } from '@sinclair/typebox';
-import {
-  ReplicasSchema,
-  PodTypeSchema,
-  IndexNameSchema,
-  DeletionProtectionSchema,
-} from './types';
-
-const ConfigureIndexOptionsSchema = Type.Object(
-  {
-    spec: Type.Optional(
-      Type.Object({
-        pod: Type.Object({
-          replicas: Type.Optional(ReplicasSchema),
-          podType: Type.Optional(PodTypeSchema),
-        }),
-      })
-    ),
-    deletionProtection: Type.Optional(DeletionProtectionSchema),
-  },
-  { additionalProperties: false }
-);
+import { ValidateProperties } from '../utils/validateProperties';
 
 export const configureIndex = (api: ManageIndexesApi) => {
-  const indexNameValidator = buildValidator(
-    'The first argument to configureIndex',
-    IndexNameSchema
-  );
-  const patchRequestValidator = buildValidator(
-    'The second argument to configureIndex',
-    ConfigureIndexOptionsSchema
-  );
-
-  return async (
+  const validator = async (
```

Comment:
Example of embedded `validator` function within `configureIndex` call that replaces the old `buildValidator` functionality that required `typebox` to check values against schema objs we created in `types.ts`

Reply:
It seems like these `validator` functions are all defined as `async`. I don't think they should be, these functions shouldn't be doing any kind of network requesting or work which would require `async`.